tests/components/dyson/test_climate.py

```diff
@@ -357,3 +398,337 @@ async def test_setup_component_with_parent_discovery(
     await hass.async_block_till_done()
 
     assert len(hass.data[dyson.DYSON_DEVICES]) == 2
+
+
+@asynctest.patch("libpurecool.dyson.DysonAccount.login", return_value=True)
+@asynctest.patch(
+    "libpurecool.dyson.DysonAccount.devices",
+    return_value=[_get_dyson_purehotcool_device()],
+)
+async def test_purehotcool_component_setup_only_once(devices, login, hass):
+    """Test if entities are created only once."""
+    config = _get_config()
+    await async_setup_component(hass, dyson_parent.DOMAIN, config)
+    await hass.async_block_till_done()
+    discovery.load_platform(hass, "climate", dyson_parent.DOMAIN, {}, config)
+    await hass.async_block_till_done()
+
+    climate_devices = [
+        device
+        for device in hass.data[DOMAIN].entities
+        if device.platform.platform_name == dyson_parent.DOMAIN
+    ]
+
+    assert len(climate_devices) == 1
+    assert climate_devices[0].serial == "XX-XXXXX-XX"
```

Comment:
Please remove this. We just want to assert the state of the entity and that not duplicate entities are created. Check that via the core state machine, `hass.states`.